agent/consul/auto_encrypt.go

```diff
@@ -59,7 +59,7 @@ func (c *Client) RequestAutoEncryptCerts(servers []string, defaultPort int, toke
 	}
 
 	// Create a CSR.
-	csr, err := connect.CreateCSR(id, pk)
+	csr, err := connect.CreateCSR("Consul RPC", id, pk)
```

Comment:
@i0rek what do you think about this? We now are adding a Common name to certs because some CA providers require them but is having every client agent in the cluster have the same CN like this is OK? Their spiffe IDs are all distinct so it feels odd that any tooling that only lists CN if present would make it look like they are all the same? Maybe we should make a fake DNS-like format for each client that involves the unique node ID, something like `client-<UUID>.consul`?

Reply:
I like the idea of a unique CN per agent certificate.

Having that said, if I am not mistake the CN is empty right now and thus the same for every agent.